packages/docprovider-extension/src/index.ts

```diff
@@ -17,14 +17,19 @@ import {
   WebSocketProviderWithLocks
 } from '@jupyterlab/docprovider';
 import { ServerConnection } from '@jupyterlab/services';
+import { IStateDB } from '@jupyterlab/statedb';
 
 /**
  * The default document provider plugin
  */
 const docProviderPlugin: JupyterFrontEndPlugin<IDocumentProviderFactory> = {
   id: '@jupyterlab/docprovider-extension:plugin',
+  requires: [IStateDB],
```

Comment:
How about making `IStateDB` an `optional` dependency, so the data is persisted only if the service is available?

Reply:
Something like this:

```typescript
...
optional: [IStateDB],
activate: (
  app: JupyterFrontEnd,
  state: IStateDB | null
): IDocumentProviderFactory => {
...
```

Reply:
Thanks @jtpio for taking a look. Statedb made opetional here https://github.com/jupyterlab/jupyterlab/pull/10490/commits/07dbd02cf0367acd787295094f5af13e6bc5dcba (I think, let me know :) )